enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,350 @@
+encrypted with those keys--but users would expect the persistent storage volumes to
+be encrypted as well. This enhancement proposes a solution for AWS, GCP, & Azure.
+
+## Motivation
+
+The primary motivation is to enable managed services to encrypt storage volumes with
+customer-provided keys automatically as part of the installation process without requiring
+day-2 operations. The behavior of automatically setting the default storage class is
+sensible for individual (non-managed-service) users as well; setting the default will
+reduce their day-2 burden and decrease the opportunity for user error.
+
+### User Stories
+
+When I install a cluster, I want to provide user-managed encryption keys in the install-config
+so that the default storage class will be set to provision volumes with those keys.
+
+### Goals
+
+* Default storage class is set to use user-provided keys at cluster creation for public clouds (AWS, GCP, Azure)
+
+### Non-Goals
+* Other clouds besides AWS, GCP, & Azure are out of scope
+
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster creator specifies (aws|gcp|azure) user-managed encryption keys 
+in `defaultMachinePlatform` of install config.
+2. Installer adds encryption key fields to cluster infrastructure object.
+3. Cluster-storage-operator/CSI drivers use cluster infrastructure object
+to create default storage class using user-encrypted keys from infra object.
+
+
+### API Extensions
+
+This enhancement extends the cluster `Infrastructure` API object to add fields
+for the user-managed encryption keys. The fields will be added to the platform
+spec and status for each platform. They are added to the spec so that users can edit the
+spec day 2 in order to update the default storage class. The status field would
+be updated by the cluster-storage operator/CSI driver.
+
+#### AWS 
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+	// [... existing fields ...]
+
+    // The KMS key to be used in the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+
+// AWSPlatformStatus holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformStatus struct {
+    // [... existing fields ...]
+
+    // The KMS key used by the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+```
+
+#### Azure
+
+```go
+// AzurePlatformSpec holds the desired state of the Azure infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AzurePlatformSpec struct{
+    // DiskEncryptionSet describes an existing disk encryption set containing
+    // user-managed encryption keys to be used by the default storage class to
+    // encrypt volumes.
+    // DiskEncryptionSet is of the form:
```

Comment:
A validation format would be helpful here

Is this field name likely to conflict in the future if we were to add other usage of Azure KMS? Again, wondering if this should be in a storage struct with default in the name or something